### PR TITLE
Bug 1268417 - [TV][Home] Support pinning the same card multiple times…

### PR DESCRIPTION
**Pull request by luke-chang**
… to homescreen

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=295c7d3e1d9e91a7ae9316d667ad7ad03580fa70

File: shared/js/smart-screen/card_manager.js [hunk 1]
```diff
@@ -806,14 +808,18 @@
...
-      return this._getPipedPromise('isPinned', function(resolve, reject) {
-        that._asyncSemaphore.wait(function() {
-          resolve(that._reloadCardList());
-        }, that);
-      }).then(function() {
-        return Promise.resolve(!!that.findCardFromCardList(options));
-      });
+      if (CardManager.ENABLE_MULTI_PIN) {
+        return Promise.resolve(false);
+      } else {
```
Comment (rexboy7) [inline shared/js/smart-screen/card_manager.js]: Unnecessary `else` clause

File: tv_apps/smart-home/js/card_picker.js [hunk 2]
```diff
@@ -180,18 +180,27 @@
...
 
-      var keys = Object.keys(candidates);
-      if (keys.length) {
-        keys.sort();
-        keys.forEach(key => appendToGridView(candidates[key]));
+      if (candidates.length) {
+        candidates.sort((a, b) => {
+          var compare = a.key.localeCompare(b.key);
+          if (!compare) {
+            return a.index - b.index;
```
Comment (rexboy7) [inline tv_apps/smart-home/js/card_picker.js]: I don't quite sure what this tries to do, shouldn't it be `a.key` and b.key`? can you explain it?

File: shared/js/smart-screen/card_manager.js (hunk 1 above)
Comment (luke-chang) [inline shared/js/smart-screen/card_manager.js]: I intended to make this feature configurable so the `else` is for the case `ENABLE_MULTI_PIN = false`.

File: tv_apps/smart-home/js/card_picker.js (hunk 2 above)
Comment (luke-chang) [inline tv_apps/smart-home/js/card_picker.js]: `a.key` and `b.key` is compared at line 197. This line is to make the sort stable.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bb4a7e90dcaab8f0e774f22f19bda8a0c0dcf9d2

**Pull request merged**